src/components/stepforms/widgets/DateComponents/DateRangeInput.vue

```diff
@@ -0,0 +1,341 @@
+ */
+@Component({
+  name: 'date-range-input',
+  components: {
+    CustomVariableList,
+    Popover,
+    Tabs,
+    RangeCalendar,
+    RelativeDateRangeForm,
+  },
+})
+export default class DateRangeInput extends Vue {
+  @Prop({ default: '' })
+  value!: string | CustomDateRange;
+
+  @Prop({ default: () => [] })
+  availableVariables!: VariablesBucket;
+
+  @Prop({ default: () => [] })
+  relativeAvailableVariables!: VariablesBucket; // variables to use in RelativeDateRangeForm "from"
+
+  @Prop({ default: () => ({ start: '', end: '' }) })
+  variableDelimiters!: VariableDelimiters;
+
+  isEditorOpened = false;
+  isEditingCustomVariable = false; // force to expand custom part of editor
+  alignLeft: string = POPOVER_ALIGN.LEFT;
+  selectedTab = 'Dynamic';
+
+  get tabs(): string[] {
+    return ['Dynamic', 'Fixed'];
+  }
+
+  // keep each tab value in memory to enable to switch between tabs without loosing content
+  tabsValues: Record<string, CustomDateRange> = {
+    Fixed: { start: new Date() },
+    Dynamic: { date: '', quantity: -1, duration: 'year' },
+  };
+
+  get currentTabValue(): CustomDateRange {
+    return this.tabsValues[this.selectedTab];
+  }
+
+  set currentTabValue(value: CustomDateRange) {
+    this.tabsValues[this.selectedTab] = value;
+  }
+
+  get variable(): AvailableVariable | undefined {
+    if (typeof this.value !== 'string') return undefined;
+    const identifier = extractVariableIdentifier(this.value, this.variableDelimiters);
+    return this.availableVariables.find(v => v.identifier === identifier);
+  }
+
+  get selectedVariables(): string {
+    // needed to select the custom button in CustomVariableList
+    if (this.isCustom) {
+      return 'custom';
+    } else {
+      return this.variable?.identifier ?? '';
+    }
+  }
+
+  get hasCustomValue(): boolean {
+    // value is custom if not undefined and not a preset variable
+    return (this.value && !this.variable) as boolean;
+  }
+
+  get isCustom(): boolean {
+    return this.hasCustomValue || this.isEditingCustomVariable;
+  }
+
+  get isFixedTabSelected(): boolean {
+    return this.selectedTab === 'Fixed';
+  }
+
+  get label(): string {
+    if (this.variable) {
+      return this.variable.label;
+    } else if (isDateRange(this.value)) {
+      return dateRangeToString(this.value);
+    } else if (isRelativeDateRange(this.value)) {
+      return relativeDateRangeToString(
+        this.value,
+        this.relativeAvailableVariables,
+        this.variableDelimiters,
+      );
+    } else {
+      return 'Select a date';
+    }
+  }
+
+  created() {
+    this.initTabs();
+  }
+
+  // init tabs by selecting correct tab and value based on prop value
+  initTabs(): void {
+    if (isDateRange(this.value)) {
+      this.tabsValues.Fixed = this.value;
+      this.selectTab('Fixed');
+    } else if (isRelativeDateRange(this.value)) {
+      this.tabsValues.Dynamic = this.value;
+      this.selectTab('Dynamic');
+    }
+    // TOFIX
```

Comment:
What should be fixed ?

Reply:
:100: Ah on dirait un oubli :sweat_smile:

Reply:
https://github.com/ToucanToco/weaverbird/pull/1011/commits/cbde8664783fc2f76e4869e69ee5250e39014fc8